flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/JobReconciler.java

```diff
@@ -101,12 +104,20 @@ public JobReconciler(
                     restoreFromLastSavepoint(flinkApp, effectiveConfig);
                 } else if (upgradeMode == UpgradeMode.LAST_STATE) {
                     final String savepointLocation =
-                            flinkApp.getStatus().getJobStatus().getSavepointLocation();
+                            flinkApp.getStatus()
+                                    .getJobStatus()
+                                    .getSavepointInfo()
+                                    .getLastSavepoint()
+                                    .getLocation();
                     // Upgrade mode changes from savepoint -> last-state
                     deployFlinkJob(
                             flinkApp, effectiveConfig, Optional.ofNullable(savepointLocation));
                 }
             }
+            ReconciliationUtils.updateForSpecReconciliationSuccess(flinkApp);
+        } else if (SavepointUtils.shouldTriggerSavepoint(flinkApp)) {
```

Comment:
IMO, This should be one of the sub-condition of `specChanged`. Please correct me if I'm wrong.

Reply:
We cannot take a savepoint while reconciling the cluster, hence it is separated to another reconcile loop.